tests/integration/components/sl-radio-group-test.js

```diff
@@ -0,0 +1,165 @@
+import { moduleForComponent, test } from 'ember-qunit';
+import hbs from 'htmlbars-inline-precompile';
+
+moduleForComponent( 'sl-radio-group', 'Integration | Component | sl radio group', {
+    integration: true
+});
+
+test( 'The disabled state applies the disabled attribute and class', function( assert ) {
+    this.set( 'testName', 'test' );
+    this.render( hbs`
+        {{#sl-radio-group disabled=true name='testName'}}
+        {{/sl-radio-group}}
+    ` );
+
+    assert.equal(
+        this.$( '.sl-radio-group' ).attr( 'disabled' ),
+        'disabled',
+        'has "disabled" attribute'
+    );
+
+    assert.ok(
+        this.$( '.sl-radio-group' ).hasClass( 'disabled' ),
+        'has "disabled" class'
+    );
+});
+
+test( 'The disabled state applies to sl-radio children', function( assert ) {
+    this.set( 'testName', 'test' );
+    this.render( hbs`
+        {{#sl-radio-group disabled=true name='testName'}}
+            {{#sl-radio label="One" value="one"}}
+            {{/sl-radio}}
+            {{#sl-radio label="Two" value="two"}}
+            {{/sl-radio}}
+            {{#sl-radio label="Three" value="three"}}
+            {{/sl-radio}}
+        {{/sl-radio-group}}
+    ` );
+
+    assert.equal(
+        this.$( '.sl-radio.disabled' ).length,
+        3,
+        'Rendered component has three disabled radio buttons'
+    );
+
+    assert.equal(
+        this.$( '.sl-radio input[disabled]' ).length,
+        3,
+        'Rendered component has three disabled inputs'
+    );
+});
+
+test( 'The readonly state applies to sl-radio children', function( assert ) {
+    this.set( 'testName', 'test' );
+    this.render( hbs`
+        {{#sl-radio-group disabled=true name='testName' readonly=true}}
+            {{#sl-radio label="One" value="one"}}
+            {{/sl-radio}}
+            {{#sl-radio label="Two" value="two"}}
+            {{/sl-radio}}
+            {{#sl-radio label="Three" value="three"}}
+            {{/sl-radio}}
+        {{/sl-radio-group}}
+    ` );
+
+    assert.equal(
+        this.$( '.sl-radio.readonly' ).length,
```

Comment:
`.sl-radio` seems more specific than necessary, and also arbitrary as there are other classes applied to the component as well.